### PR TITLE
Fix query leadership-schedule command.

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: cardano-api/src/Cardano/Api/LedgerState.hs [hunk 1]
```diff
@@ -1352,22 +1356,22 @@ nextEpochEligibleLeadershipSlots sbe sGen serCurrEpochState _ptclState
...
-  let TPraos.PrtclState _ evolvingNonce candidateNonce = TPraos.csProtocol chainDepState
+  let Consensus.PraosNonces { Consensus.candidateNonce, Consensus.evolvingNonce } =
+        Consensus.getPraosNonces (Proxy @(ConsensusProtocol era)) chainDepState
 
   when (evolvingNonce == candidateNonce)
    $ Left LeaderErrCandidateNonceStillEvolving
 
   -- Get the previous epoch's last block header hash nonce
-  let Tick.TicknState _ prevEpochLastBlkHeaderHashNonce = TPraos.csTickn chainDepState
+  let previousLabNonce = Consensus.previousLabNonce (Consensus.getPraosNonces (Proxy @(ConsensusProtocol era)) chainDepState)
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/LedgerState.hs:1372]: Please check.

**Comment by newhoggy** [comment]
Example output for `current`:

```
$ CARDANO_NODE_SOCKET_PATH=example/node-spo1/node.sock cardano-cli query leadership-schedule --testnet-magic 42 --genesis example/genesis/shelley/genesis.json --stake-pool-id $(cardano-cli stake-pool id --cold-verification-key-file example/pools/cold1.vkey --output-format "hex") --vrf-signing-key-file example/node-spo1/vrf.skey --current
     SlotNo                          UTC Time
-------------------------------------------------------------
     544                   2022-05-30 14:20:39.227905 UTC
     548                   2022-05-30 14:20:39.627905 UTC
     551                   2022-05-30 14:20:39.927905 UTC
     594                   2022-05-30 14:20:44.227905 UTC
     618                   2022-05-30 14:20:46.627905 UTC
     632                   2022-05-30 14:20:48.027905 UTC
     674                   2022-05-30 14:20:52.227905 UTC
     704                   2022-05-30 14:20:55.227905 UTC
     708                   2022-05-30 14:20:55.627905 UTC
     754                   2022-05-30 14:21:00.227905 UTC
     773                   2022-05-30 14:21:02.127905 UTC
     790                   2022-05-30 14:21:03.827905 UTC
     837                   2022-05-30 14:21:08.527905 UTC
     841                   2022-05-30 14:21:08.927905 UTC
     858                   2022-05-30 14:21:10.627905 UTC
     866                   2022-05-30 14:21:11.427905 UTC
     889                   2022-05-30 14:21:13.727905 UTC
     957                   2022-05-30 14:21:20.527905 UTC
     962                   2022-05-30 14:21:21.027905 UTC
     982                   2022-05-30 14:21:23.027905 UTC
```

**Comment by newhoggy** [comment]
Example output for `next`:

```
$ CARDANO_NODE_SOCKET_PATH=example/node-spo1/node.sock cardano-cli query leadership-schedule --testnet-magic 42 --genesis example/genesis/shelley/genesis.json --stake-pool-id $(cardano-cli stake-pool id --cold-verification-key-file example/pools/cold1.vkey --output-format "hex") --vrf-signing-key-file example/node-spo1/vrf.skey --next
     SlotNo                          UTC Time
-------------------------------------------------------------
     7501                   2022-05-30 14:32:14.927905 UTC
     7508                   2022-05-30 14:32:15.627905 UTC
     7546                   2022-05-30 14:32:19.427905 UTC
     7554                   2022-05-30 14:32:20.227905 UTC
     7634                   2022-05-30 14:32:28.227905 UTC
     7655                   2022-05-30 14:32:30.327905 UTC
     7669                   2022-05-30 14:32:31.727905 UTC
     7672                   2022-05-30 14:32:32.027905 UTC
     7709                   2022-05-30 14:32:35.727905 UTC
     7720                   2022-05-30 14:32:36.827905 UTC
     7766                   2022-05-30 14:32:41.427905 UTC
     7776                   2022-05-30 14:32:42.427905 UTC
     7827                   2022-05-30 14:32:47.527905 UTC
     7885                   2022-05-30 14:32:53.327905 UTC
     7908                   2022-05-30 14:32:55.627905 UTC
     7937                   2022-05-30 14:32:58.527905 UTC
```

File: cardano-api/src/Cardano/Api/LedgerState.hs (hunk 1 above)
Comment (nfrisby) [reply cardano-api/src/Cardano/Api/LedgerState.hs:1372]: We've had

```
data TicknState = TicknState
  { ticknStateEpochNonce :: !Nonce,
    ticknStatePrevHashNonce :: !Nonce
  }
```

and now we also have

```
instance SL.PraosCrypto c => PraosProtocolSupportsNode (TPraos c) where
  getPraosNonces _prx cdst =
      PraosNonces {
          candidateNonce
        , epochNonce       = ticknStateEpochNonce
        , evolvingNonce
        , labNonce         = csLabNonce
        , previousLabNonce = ticknStatePrevHashNonce
        }
    where
      TPraosState { tpraosStateChainDepState } = cdst
      SL.ChainDepState {
          SL.csLabNonce
        , SL.csProtocol
        , SL.csTickn
        } = tpraosStateChainDepState
      SL.PrtclState
        _opcertCounters
        evolvingNonce
        candidateNonce
          = csProtocol
      SL.TicknState {
          ticknStateEpochNonce
        , ticknStatePrevHashNonce
        } = csTickn
```

so, yes, I think `Consensus.previousLabNonce` is indeed the new name of the second argument of `TicknState`.

**Review by nfrisby — COMMENTED** [review]
I replied to the question about the new name for the nonce. Thanks for the ping, @newhoggy !

File: cardano-api/src/Cardano/Api/LedgerState.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/LedgerState.hs:1372]: Thanks!

**Pull request merged**